src/libcore/ptr.rs

```diff
@@ -10,9 +10,29 @@
 
 // FIXME: talk about offset, copy_memory, copy_nonoverlapping_memory
 
-//! Raw, unsafe pointers, `*const T`, and `*mut T`.
+//! Manually manage memory through raw pointers.
 //!
 //! *[See also the pointer primitive types](../../std/primitive.pointer.html).*
+//!
+//! # Safety
+//!
+//! Many functions in this module take raw pointers as arguments and dereference
+//! them. For this to be safe, these pointers must be valid. A valid pointer
+//! is one that satisfies **all** of the following conditions:
```

Comment:
Validity of a pointer for an access is a property of the pointer, the size of the access and the kind of the access (reading or writing). That should be taken into account both by this definition and by every place that references it.

Reply:
Interesting, the old PR used to differentiate between reads and writes, but only by requiring that reads must be done from initialized memory. I had to remove this, because it's not something we're ready to commit to, although it's currently suggested by the reference. I should have split validity into "valid for reads" and "valid for writes".

I view the size of the access as defined by the pointer type, so a `*const u32` can be valid for reads,(implying that the pointed-to region of memory is at least four bytes long), and/or valid for writes.